[T09-4] Car Park Finder

dlworldpeace opened the pull request:
@YurouTang @ZhuanYu @Lexzt @Alvinnyk

eugenepeh commented on README.adoc:
```diff
@@ -4,7 +4,6 @@ ifdef::env-github,env-browser[:relfileprefix: docs/]
 https://travis-ci.org/se-edu/addressbook-level4[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
 https://ci.appveyor.com/project/damithc/addressbook-level4[image:https://ci.appveyor.com/api/projects/status/3boko2x2vr5cc3w2?svg=true[Build status]]
 https://coveralls.io/github/se-edu/addressbook-level4?branch=master[image:https://coveralls.io/repos/github/se-edu/addressbook-level4/badge.svg?branch=master[Coverage Status]]
-https://www.codacy.com/app/damith/addressbook-level4?utm_source=github.com&utm_medium=referral&utm_content=se-edu/addressbook-level4&utm_campaign=Badge_Grade[image:https://api.codacy.com/project/badge/Grade/fc0b7775cf7f4fdeaf08776f3d8e364a[Codacy Badge]]
```
can update with your own repo link instead if you've setted up these CI services.

eugenepeh reviewed (commented):
Good start on setting up the repository. 

Do continue working towards v1.1.

eugenepeh commented:
Just another quick note, do enable issue tracker on your team's repo and ensure that your travis pass the CI test.

Also remember to check https://nus-cs2103-ay1819s1.github.io/cs2103-website/ on all the v1.1 to-dos

eugenepeh commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -1,61 +1,58 @@
+    //        + PREFIX_EMAIL + "johnd@example.com "
+    //        + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
+    //        + PREFIX_TAG + "friends "
+    //        + PREFIX_TAG + "owesMoney";
```
dead code can be removed.

eugenepeh commented on src/main/java/seedu/address/logic/commands/ClearCommand.java:
```diff
@@ -14,7 +14,6 @@
     public static final String COMMAND_WORD = "clear";
     public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
 
-
```
Be careful of unnecessary or accidental changes here as it can cause unnecessary merge conflict.

eugenepeh commented on src/main/java/seedu/address/logic/parser/AddCommandParser.java:
```diff
@@ -31,22 +43,35 @@
+        NightParking nightParking = ParserUtil.parseNightParking(argMultimap.getValue(PREFIX_NIGHT_PARK).get());
+        ShortTerm shortTerm = ParserUtil.parseShortTerm(argMultimap.getValue(PREFIX_SHORT_TERM).get());
+        TotalLots totalLots = ParserUtil.parseTotalLots(argMultimap.getValue(PREFIX_TOTAL_LOTS).get());
+        TypeOfParking typeOfParking = ParserUtil.parseTypeOfParking(argMultimap.getValue(PREFIX_TYPE_PARK).get());
```
Seems like a huge string to type in a go!

eugenepeh commented on src/main/java/seedu/address/model/carpark/FreeParking.java:
```diff
@@ -0,0 +1,57 @@
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a car park's free parking in the address book.
```
address book? :-P

eugenepeh reviewed (commented):
A quick look through at your code. 
Generally looks okay, dead code should be removed or updated.

Great Work!

Other things:
* Gh-pages
   * Good!
* README
   * Mock-up looks good
* User-guide 
   * looks good
   * features not implemented yet should be clearly marked as Coming in v2.0
   * perhaps can add back the command summary? it helps the user to know all the commands in quick look.
* Developer guide 
   * You can remove or update Appendix F if you want
   * _{More to be added}_  can be removed if no longer applicable
   * NFR
      * `The car park data should be updated frequently while running to send notifications in real time.`
         sounds like a feature to me, maybe give a dead line or something

eugenepeh commented:
@ZhuanYu @dlworldpeace @YurouTang @Lexzt @Alvinnyk
## Feedback for v1.2 T09-4

### Developer Guide
* Could do with more explanation,
![image](https://user-images.githubusercontent.com/19277206/47276581-b4f7c380-d5ea-11e8-9e0b-f5561ebb2ff2.png)

* Why is there so many decision node? Also take note that the final activity node suppose to look different from start node.
![image](https://user-images.githubusercontent.com/19277206/47276677-63036d80-d5eb-11e8-9d9e-878136e24901.png)

* 3.6. Query feature
Does it save a local copy at the end? If so, do highlight that too.

* Note that the activity bar suppose to end the moment it returns
![image](https://user-images.githubusercontent.com/19277206/47276749-0785af80-d5ec-11e8-9a43-c88b391f024a.png)


### Overall
* Progress seems to be good, be sure to make it user testable soon!
* For the design considerations, try to think of other pros instead of simply because its easy to implement.

ZhuanYu committed:
Remove unused imports

Lexzt committed:
Merge pull request #82 from ZhuanYu/master

Updates on Notify and in general

Lexzt committed:
Almost done with postal code handling.
Pulling data from Onemap to convert x/y coorindate to postal code.
Updated fxml to deal with postal code.
Changed the carpark constructor to take in postal code.
Updated sample data with sample postal code.
Introduced postal code data in txt file.

Lexzt committed:
Fixed test cases
Resolve conflict and merge codes

Lexzt committed:
Fixing test cases for postal code by updating the xmls.

Lexzt committed:
Updated carpark list card with postal code.

Lexzt committed:
Updated dummy search page

ZhuanYu committed:
Merge pull request #80 from YurouTang/master

Improved Filter command

ZhuanYu committed:
Minor changes to query feature in developer guide

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Fixed CSS warning

dlworldpeace committed:
Update DG. change 'Tab' key to kbd:[Tab].

YurouTang committed:
Defensive coding + Restructure of Filter Command

Lexzt committed:
Adding test cases for Gson.

Lexzt committed:
Removed unused imports

ZhuanYu committed:
Merge pull request #83 from Lexzt/master

Postal Code and documentation

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

YurouTang committed:
Added more flags to filter function
- availability of parking slots
- short term parking
- type of parking system

dlworldpeace committed:
Change diagram wording to carparkfinder instead of addressbook.

Lexzt committed:
Updating DG diagrams

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Re added redo/undo and updated photos for it accordingly.

Lexzt committed:
Merge pull request #84 from dlworldpeace/master

Update DG to meet milestone 1.3

Lexzt committed:
Merging docs

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Updated with annotations for images

Lexzt committed:
Updated more photos, sequence diagrams.

Lexzt committed:
Cleared up undo/redo Design Considerations for table format

Lexzt committed:
Cleanup of english for developer guide

Lexzt committed:
Added basic documentation for developer guide

Lexzt committed:
Clean up, remove unused lines

Lexzt committed:
Removed css, put the stylesheet directly in html

ZhuanYu committed:
Merge pull request #85 from Lexzt/master

Updated developer guide formatting

YurouTang committed:
Filter Command DG updated

dlworldpeace committed:
Change alias to abbreviation in the entire project.

dlworldpeace committed:
Replace if else if with switch in CommandBox.java

dlworldpeace committed:
Add test cases for command abbreviation feature. It is tested through parser.

ZhuanYu committed:
Merge pull request #86 from dlworldpeace/master

Add test cases for command abbreviation parsing. Replace all 'alias' in the project with 'abbreviation'.

ZhuanYu committed:
Improve on Query and Notify commands.

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

dlworldpeace committed:
Link keywords in UG to its designated section.

dlworldpeace committed:
Change diagram for quick start to make it clearer for user to understand.

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
Merge pull request #87 from dlworldpeace/master

Update UG to make it easier to comprehend, especially Quick Start Section.

ZhuanYu committed:
Change to to v1.3 in main app

Add new NoSelectionRequestEvent.java to handle deselect of car park card.
Add newline for DataFetchExceptionEvent.java
Add check for CarparkFinderParser.java where if the command has no arguements it should throw ParseException.
Remove unnccessary imports in GsonUtil.java, QueryCommand.java

Add new method isNonNegativeAboveNineInteger(). This helps to check if seconds for NotifyCommand is within the range of 10 to 600 and also allow 0 to turn it off.
Add post event in ClearCommand to deselect the car park card when there are no more car parks.
Add newline in FindCommand so that the MESSAGE_USAGE is not that long.
Move the logic for NotifyCommand to the new NotifyTimeTask.java, which will create a separate thread to handle the timer task.
Change the check for parseTime to NonNegativeAboveNineInteger() method in ParserUtil.java
NotifyCommand and QueryCommand now works using multithreading.
Add Timer and NoSelectionRequestEvent to CarparkListPanel.java
Enablle the Command box when exception is done
Make the Message Box bigger

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
Remove check for arguements for commands that don't have it
Fix UserGuide Filter to Find under find section

Lexzt committed:
Merge pull request #88 from ZhuanYu/master

Updated Notify and Query for v1.3

ZhuanYu committed:
Changes to UG for query and notify

Lexzt committed:
Updated HTML with new map.
Updated documentation for friday project portfolio.
Updated list command to display all car park in the map.
Introduced ListCarparkRequestEvent to display all the car park in the google map.

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Updating json for private access issue.

Lexzt committed:
Fixing html style check errors

Lexzt committed:
FIxing html checkstyle error 2

Lexzt committed:
Testing checkstyle formatting for html

Lexzt committed:
Fixing ascii doc errors again 3

Lexzt committed:
Finally removed all trailing white spaces and extra lines.

YurouTang committed:
Updated and tidied User Guide

YurouTang committed:
Merge branch 'master' into master

Lexzt committed:
Fixing test cases, finally should all pass now. Test cases had issues due to changing of html

ZhuanYu committed:
Merge pull request #89 from Lexzt/master

HTML and List command updates

YurouTang committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

# Conflicts:
#	docs/UserGuide.adoc
#	src/main/java/seedu/parking/model/carpark/CarparkFilteringPredicate.java

YurouTang committed:
Merge branch 'master' of https://github.com/YurouTang/main

# Conflicts:
#	src/main/java/seedu/parking/model/carpark/CarparkFilteringPredicate.java

YurouTang committed:
Fixed checkstyle

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

dlworldpeace committed:
Fix missing images in AboutUs.adoc

ZhuanYu committed:
Merge pull request #90 from YurouTang/master

Many new stuffs... including update of UGDG

ZhuanYu committed:
Merge pull request #91 from dlworldpeace/master

Fix missing images in AboutUs.adoc

ZhuanYu committed:
Updates to documentation for v1.3

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc

ZhuanYu committed:
Update picture

Lexzt committed:
Merge pull request #92 from ZhuanYu/master

v1.3

ZhuanYu committed:
Updates to fix v1.3

Lexzt committed:
Merge pull request #93 from ZhuanYu/master

Updates to fix v1.3

YurouTang committed:
Edits to UGDG with Calculate command DG still in progress

YurouTang committed:
Updated Project Portfolio

ZhuanYu committed:
WIP for PPP

YurouTang committed:
Merge branch 'master' into master

YurouTang committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc

Lexzt committed:
Updated undo/redo for user guide, and updated personal project portfolio.

dlworldpeace committed:
Update personal Project Portfolio up to v1.3

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Merge pull request #95 from YurouTang/master

Edits to documentation

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Merge pull request #97 from dlworldpeace/master

Update PPP up to v1.3

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
Merge pull request #98 from Lexzt/master

Merging Delong and Yurou PPP.

Lexzt committed:
Updated more docs, photo, personal details

ZhuanYu committed:
Merge pull request #99 from Lexzt/master

Updated more docs, photo, personal details

ZhuanYu committed:
Bug fix for QueryCommand not showing the list of car parks again if query were to fail.
Updated my PPP.

ZhuanYu committed:
Merge branch 'master' into master

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc

ZhuanYu committed:
Add back tag

ZhuanYu committed:
Merge branch 'master' of https://github.com/ZhuanYu/main

# Conflicts:
#	docs/DeveloperGuide.adoc

ZhuanYu committed:
Comment out test case for postal code since google token expired.

ZhuanYu committed:
Commented out unused imports for GsonUtilTest.java.
Fix error "only book doctypes can contain level 0 sections" for keith.adoc and delong.adoc

Lexzt committed:
Merge pull request #96 from ZhuanYu/master

Updates to PPP and Bug fix

ZhuanYu committed:
Trying to see if parseInt fixes the cutting of markers
Mini update to my PPP

Lexzt committed:
Merge pull request #101 from ZhuanYu/master

Trying to see if parseInt fixes the cutting of markers

Lexzt committed:
Testing ascii doc pdf exporter

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Updated documentation for PPP
Updated HTML/JS/CSS for displaying new filtered/find command
Updated HTML to extend according to marker.

Lexzt committed:
Fixing checkstyle error for HTML,CSS and personal adoc

Lexzt committed:
Fixing test cases due to the change of the output of cards. Failing one more, work on it tmr

YurouTang committed:
Updated ppp

YurouTang committed:
Updated PPP with styles

Lexzt committed:
Fixing test cases, and checkstyle error resulting from different html handling.

Lexzt committed:
Fixing some coveralls issues

Lexzt committed:
Removed unused imports

ZhuanYu committed:
Add markerCluster to google map

ZhuanYu committed:
Remove extra lines

Lexzt committed:
Merge pull request #103 from ZhuanYu/master

Add markerCluster to google map

Lexzt committed:
Fixed the HTML, moved the js to an online server temp, and moved posted code into resources folder for future implementation

ZhuanYu committed:
Merge pull request #102 from Lexzt/master

Updated documentation and UI

ZhuanYu committed:
Changes to PPP but halfway

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
Revert "Changes to PPP but halfway"

This reverts commit d832bf740b4ee3262c2829ddd5db0f9907c03b64.

ZhuanYu committed:
Rename DummySearchPage.html to MainPage.html
Remove stylesheet being set to gh-pages.css in build.gradle
Remove gh-pages.css overrides, leaving it with only the header section

Add stylesheet default.css to all asciidoc that requires it
Add zhuanyu.css for my PPP
Change sections of DG and UG for my PPP

Change limit and offset for pulling of data to 2000 in GsonUtil.java and MainPage.html
Change condition in while loop to remove the extra call to the database to check for no more data

Change MainPage.html to point towards a local version
Add a total number of car parks in Car Park Finder at the centre of the footer
Reorder UI for MainWindow.fxml

Add two new javascript files: svy21.js and markerclusterer.js
Add cluster marker icons locally

ZhuanYu committed:
Fix checkstyle errors

ZhuanYu committed:
Rename

ZhuanYu committed:
Rename method naming convention for getReturnList()
Fix test cases for Main Page

ZhuanYu committed:
Remove unused import

ZhuanYu committed:
Revert changing MainPage to local and remove script files.

ZhuanYu committed:
Change to MAIN_PAGE

Lexzt committed:
Merge pull request #113 from ZhuanYu/master

v1.3.1 Update PPP for all

ZhuanYu committed:
change markers path location

Lexzt committed:
Merge pull request #118 from ZhuanYu/master

change markers path location

ZhuanYu committed:
v1.3.1

Lexzt committed:
Merge pull request #119 from ZhuanYu/master

v1.3.1

Lexzt committed:
Fixing index adoc for change of style.
Fixed my own PPP and updated my own documentations.

Lexzt committed:
Updating user docs
Updated html to accept 0 inputs and show an empty singapore

Lexzt committed:
EOF for stylesheet file checkstyle error

Lexzt committed:
Updated UserGuide with some syntax errors.
Updating clear command due to change of HTML handling to fix test cases

Lexzt committed:
Updating developer guide for UI/HTML/CSS/JS

Lexzt committed:
Updated more details for developer guide
Minor changes to stylesheet

ZhuanYu committed:
Merge pull request #120 from Lexzt/master

Updating documentation for new UI

Lexzt committed:
Fixed missing personal portfolio link in about us
Updated css for toc header
Updated adoc with some missing photo name

ZhuanYu committed:
Change the fixed size of the car park card to 280.0 from 240.0
Fix "Enter comand here..." not appearing after using Query Command.
Change the colour theme for the application
Minor changes to the padding in MainWindow.fxml

ZhuanYu committed:
Remove traces of tag at default load

ZhuanYu committed:
Rename PPP for the rest as it was not linked properly
Remove unncessary icons
Change the red marker icon to green

Change the colour scheme to a more lighter one for the application
Adjust the padding for ListPanel and the Card
Change the default size to be a smaller ratio for the application

ZhuanYu committed:
Change empty cell colour

ZhuanYu committed:
Rename DarkTheme to AppTheme
Rmove the word App from appTitle
Update colour while hovering menu

dlworldpeace committed:
Update PPP by delong.

dlworldpeace committed:
Add delong.css to delong's ppp

dlworldpeace committed:
link aboutus.adoc to each team member's ppp.

dlworldpeace committed:
label sort as coming in v2.0

dlworldpeace committed:
Change command_alias to command_abbreviation in calculateCommand

dlworldpeace committed:
Update AutoComplete mechanism to fit the new abbreviation mechanism.

Lexzt committed:
Merge pull request #122 from ZhuanYu/master

Update the colour of the application

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

dlworldpeace committed:
change test for abbreviation.

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

YurouTang committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc
#	docs/team/yuroutang.adoc

YurouTang committed:
Deleted html file

Lexzt committed:
Merge pull request #124 from YurouTang/master

Updated PPP

Lexzt committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

YurouTang committed:
Updated Filter and Calculate screenshots as feedbacked by Issues

dlworldpeace committed:
update filter command format.

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Updated README with new badge for MIT
Updated MainPage to only have one function to parse information around.
Updated test cases
Fixed and re implemented postal code functionality.
Updated Query command to now refresh the page
Updated default html to on start
Fixed all test cases that occured when updating code
Updated UiPart to have data folder, and for future use.

Lexzt committed:
Merge pull request #123 from dlworldpeace/master

Update PPP and AboutUs. Change Abbreviation mechanism to fit more cases.

Lexzt committed:
Merge code

ZhuanYu committed:
Merge pull request #125 from Lexzt/master

Update documentation and bug fixes

Lexzt committed:
Updating user name for about us page due to github ign

dlworldpeace committed:
Update UG for the new abbreviation mechanism.

dlworldpeace committed:
Merge pull request #126 from Lexzt/master

Updating user name for about us page due to github ign

dlworldpeace committed:
Test cases for CarparkFinderParser with greater coverage.

dlworldpeace committed:
add another test case for the last function in CarparkFinderParser.

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

Lexzt committed:
Merge pull request #127 from dlworldpeace/master

Update UG for new abbr mechanism. Improve test cases for CarparkFinderParser.

Lexzt committed:
Removing unused instance of PersonUtil
Updated all functions to remove the persons word
Changed the address book app name to car park finder

Lexzt committed:
Added more test case for XmlAdaptedCarpark

Lexzt committed:
Fixing coverage for find command

Lexzt committed:
Added back test cases and fixed them for coverage

Lexzt committed:
Adding in back more test case for coverage

Lexzt committed:
Remvoed delete command, fixed and added more test cases

ZhuanYu committed:
Merge pull request #129 from Lexzt/master

Removed all instances of person and added more test case

ZhuanYu committed:
Fix notfy message still appearing even though notification is turned off

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
For 1.3.2

dlworldpeace committed:
Update auto-complete mechanism and messages shown.

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

dlworldpeace committed:
temporary debug for auto-complete.

ZhuanYu committed:
Merge pull request #131 from dlworldpeace/master

New mechanism for auto-complete.

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

eugenepeh commented:
## Some quick feedback for DG at mid-v1.4:
* Are you guys still using CommandHistory? Not sure if its your team who have removed undo/redo feature.
![image](https://user-images.githubusercontent.com/19277206/48275427-6b2f2a00-e480-11e8-9619-2f8d9c438879.png)
* Problem areas spotted in diagrams (drawn in red) :
![image](https://user-images.githubusercontent.com/19277206/48275368-49ce3e00-e480-11e8-87bc-19beb4da5b0c.png)
![image](https://user-images.githubusercontent.com/19277206/48275527-a92c4e00-e480-11e8-88d5-afc01ba9a988.png)
* Can probably use same capitalization
![image](https://user-images.githubusercontent.com/19277206/48275595-d8db5600-e480-11e8-9b1a-99e92b0e1b73.png)

* Is this still in your app?
![image](https://user-images.githubusercontent.com/19277206/48275566-c4975900-e480-11e8-8049-fc3f532f6443.png)

dlworldpeace committed:
Display messages for auto-complete success and exceptions.

dlworldpeace committed:
Make sure auto complete works on filter command. but bot calculate command yet.

YurouTang committed:
Cleaned Up Filter Command + Test Cases

ZhuanYu committed:
Add new feature where when the application exits, it saves a local copy of the car park data.
Improve texts on Messages.java
Add a newline for TimeIntervalChangeEvent.java
Add more conditions for NotifyTimeTask and NotifyCommand, this is to prevent it from updating when it has been turned off
Add check for ClearCommand to stop notify from updating
Add check for QueryCommand to stop notify from updating
Add spacing for SelectCommand for the word carpark
Delete extra dash in ModelManager.java

Massive clean up on documentation to prepare for PPP.

ZhuanYu committed:
Fix Developer Guide

ZhuanYu committed:
Remove SampleDataUtil.java as no longer need sample data on load
Change test case in SampleDataTest.java as sample car parks do not load if no data file is found
Remove unnecessary imports for SampleDataUtil.

ZhuanYu committed:
Check to see if test will work

ZhuanYu committed:
Change back access levels

ZhuanYu committed:
Second check

ZhuanYu committed:
Update README introduction and added value proposition and main features
Further updates to PPP

YurouTang committed:
Cleaned up Calculate Command + Test Cases

YurouTang committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

# Conflicts:
#	src/main/java/seedu/parking/logic/commands/CalculateCommand.java
#	src/main/java/seedu/parking/logic/commands/FilterCommand.java

YurouTang committed:
Fixed Checkstyle

YurouTang committed:
Updated CarparkFinderParserTest

YurouTang committed:
Fixed Checkstyle

YurouTang committed:
Bug Fix: Lenient Date Format

YurouTang committed:
Merge pull request #134 from ZhuanYu/master

Fix to UGDG and notify command

YurouTang committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
Merge pull request #135 from YurouTang/master

Test Cases for Filter and Calculate

ZhuanYu committed:
Fix notify bug where cannot change selection
More updates to notify PPP

ZhuanYu committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

YurouTang committed:
Bug Fix: Filter Boundary Case (filters all carparks which has free parking time overlaping the input start and end times)

ZhuanYu committed:
Complete PPP for notify

ZhuanYu committed:
Fixing test case

YurouTang committed:
Updated UG and PPP

ZhuanYu committed:
Fix GsonUtil throwing null pointer by changing object to optional

ZhuanYu committed:
check if this works

ZhuanYu committed:
remove unused import

YurouTang committed:
Edited Screenshots

ZhuanYu committed:
Checking for null pointer

ZhuanYu committed:
test to see if this works

Lexzt committed:
Merge pull request #137 from ZhuanYu/master

 Fix notify bug where cannot change selection[Don't accept testing travis]

ZhuanYu committed:
Add query test

ZhuanYu committed:
add javadoc

Lexzt committed:
Working some documentation and PPP

ZhuanYu committed:
Add new notify test

Lexzt committed:
Updating documents and merging

ZhuanYu committed:
Fix

YurouTang committed:
Edit

YurouTang committed:
Reverse edit

Lexzt committed:
Updated docs for developer guide for manuel testing

Lexzt committed:
Updated syntax for developer guide for eof

ZhuanYu committed:
trying to up coverall

Lexzt committed:
Updating documentation for EOF

ZhuanYu committed:
hopefully this is final test

Lexzt committed:
Removed undo/redo due to page limit, plus alot of room for error

Lexzt committed:
More syntax changes

YurouTang committed:
Minor edit.

Lexzt committed:
Merge pull request #138 from ZhuanYu/master

Add query and notify test

YurouTang committed:
Edit.

dlworldpeace committed:
Add test cases for autocomplete feature.

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

ZhuanYu committed:
Merge pull request #139 from Lexzt/master

Updating documentation and PPP

ZhuanYu committed:
Merge pull request #136 from YurouTang/master

Bug Fix and and edits to PPP

dlworldpeace committed:
update UG on autocomplete and abbrev.

ZhuanYu committed:
Bump coverage

ZhuanYu committed:
fix

ZhuanYu committed:
Merge pull request #133 from dlworldpeace/master

Refine auto-complete feature to be more bug-free.

Lexzt committed:
Merge pull request #140 from ZhuanYu/master

Bump coverage

dlworldpeace committed:
update dg

dlworldpeace committed:
update ppp

dlworldpeace committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-4/main

dlworldpeace committed:
ppp last

ZhuanYu committed:
Merge pull request #141 from dlworldpeace/master

last commit
